src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -77,7 +65,11 @@ def __init__(self, *args, **kwargs):
         # Define a test ID using the test_* method name
         self.test_id = self.get_test_name()
 
-        self.test_dir = os.getenv("DAOS_TEST_LOG_DIR", "/tmp")
+        # Define a test unique temporary directory
+        self.base_test_dir = os.getenv("DAOS_TEST_LOG_DIR", "/tmp")
+        self.test_dir = os.path.join(self.base_test_dir, self.test_id)
+        if not os.path.exists(self.test_dir):
+            os.makedirs(self.test_dir)
```

Comment:
Under what condition the test directory exists? Restart server?

Reply:
Typically there should not be a condition where the unique temporary test directory would exist (maybe if you're repeating the test and the previous run hung or timed out early in `tearDown()` where the unique temporary test directory was not cleaned up.